### PR TITLE
Extending tests for converting tables

### PR DESCRIPTION
**Pull request by kneth**
Due to an embarrassing error a unit test, the bug #1695 wasn't discovered before release. Retrospectively, the unit tests are complex and need a few comments to understand what is tested.

@realm/java

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java [hunk 1]
```diff
@@ -551,7 +557,6 @@ else if (columnType == ColumnType.STRING)
                             table.setString(colIndex, 1, null);
                         else
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java]: C++ style. Missing {.

File: realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java [hunk 2]
```diff
@@ -574,9 +580,7 @@ else if (columnType == ColumnType.STRING)
                             table.setString(colIndex, 0, null);
                         else
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java]: Same

**Comment by cmelchior** [comment]
This whole file could use a style check, but I guess it doesn't have to be in this PR. If not :+1:

**Comment by beeender** [comment]
:+1:

File: realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java [hunk 3]
```diff
@@ -497,29 +511,32 @@ else if (columnType == ColumnType.STRING)
                     assertEquals(colIndex, table.getColumnIndex(columnName));
 
                     table.addEmptyRow();
-                    if (columnType == ColumnType.BINARY)
+                    if (columnType == BINARY)
```
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java]: missing `{}`

File: realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java [hunk 4]
```diff
@@ -456,30 +458,42 @@ public void testConvertToNullable() {
...
+                            break;
+                    }
                     try {
                         table.addEmptyRow();
-                        if (columnType == ColumnType.BINARY)
+                        if (columnType == BINARY) {
                             table.setBinaryByteArray(colIndex, 1, null);
-                        else if (columnType == ColumnType.STRING)
+                        }
+                        else if (columnType == STRING) {
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java]:
```
}
else if ()
```

should be 

```
} else if () {
``

**Comment by cmelchior** [comment]
retest this please

**Comment by dalinaum** [comment]
Besides a minor brace :+1:

File: realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java [hunk 5]
```diff
@@ -369,16 +371,16 @@ public void testShouldThrowWhenSetIndexOnWrongColumnType() {
 
     public void testColumnName() {
         Table t = new Table();
-        try { t.addColumn(ColumnType.STRING, "I am 64 characters.............................................."); fail("Only 63 characters supported"); } catch (IllegalArgumentException e) { }
-        t.addColumn(ColumnType.STRING, "I am 63 characters.............................................");
+        try { t.addColumn(STRING, "I am 64 characters.............................................."); fail("Only 63 characters supported"); } catch (IllegalArgumentException e) { }
```
Comment (nhachicha) [inline realm/realm-library/src/androidTest/java/io/realm/internal/JNITableTest.java]: *ignored* instead of *e*

**Comment by nhachicha** [comment]
:+1:

**Pull request merged**